cirq/contrib/tpu/circuit_to_tensorflow.py

```diff
@@ -0,0 +1,469 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Callable, Dict, List, NamedTuple, Tuple, TypeVar, Union
+
+from collections import namedtuple
+
+import numpy as np
+import tensorflow as tf
+
+from cirq import ops, circuits, linalg, protocols, decompositions
+
+
+# We logically divide the qubits into groups of this size, and operate on each
+# group as a multi-qubit register. This significantly improves compilation
+# times, and avoids padding overhead when the tensors are laid out into memory.
+# This is an internal detail not exposed to callers (e.g. we reshape before
+# returning).
+_GROUP_SIZE = 7
+
+
+ComputeFuncAndFeedDict = NamedTuple(
+    'ComputeFuncAndFeedDict',
+    [
+        ('compute_func', Callable[[], tf.Tensor]),
```

Comment:
nit: compute isn't quite correct, it's really doing a simulation.  In tensorflow you should call your return tensors by what they do, "compute_func" is almost "func_func".

Reply:
Renamed to 'compute'